### ISSUE TITLE
Change coordinates to reflectometry coordinate system

### ISSUE THREAD

**Issue by arm61** [scipp/essreflectometry#14]
Currently an internal (or maybe NeXus) standard is used for what is `x`, `y`, and `z`. This doesn't work for reflectometry where there is already an accepted standard that depends on the sample orientation (see https://orsopy.readthedocs.io/en/latest/orsopy.fileio.base.html#orsopy.fileio.base.ValueVector). The Amor code should be modified to adhere to this reflectometry standard.

**Comment by jokasimr** [comment]
Yes it is still relevant. But I'm not sure if we want to do anything about it.

Essreflectometry uses a global coordinate system similar to the rest of the instrument workflows. That is, sample is origin, z is the 'beam direction' (this becomes fuzzy when it comes to instruments with focusing like amor and estia, but it's basically the vector from sample to the the detector, projected to the horizontal plane), y is up, and x is whatever is left.

**Comment by SimonHeybrock** [comment]
Is there a way to avoid referring to x,y, z at all, e.g., by callings things `up`, `sample_normal`, etc.?

**Comment by jl-wynen** [comment]
This can mask some differences. But there is a fundamental difference between reflectometry and the other techniques. Normally, we define 
- one vector along beam
- one vector along gravity
- one vector in terms of first three to make a right-handed orthogonal system
But in reflectometry it seems that people use
- one vector along beam
- one vector normal to the sample
- one vector based on the other two
In this setup, you don't get an orthogonal coordinate system.
